dynamic_rest/viewsets.py

```diff
@@ -0,0 +1,31 @@
+from rest_framework import viewsets, response
+
+class DynamicModelViewSet(viewsets.ModelViewSet):
+
+  def _get_request_fields(self):
+    request_fields = self.request.QUERY_PARAMS.getlist('fields[]')
+    field_map = {}
+    for field in request_fields:
+      if field.startswith('-'):
+        field = field[1:]
+        include = False
+      else:
+        include = True
+      field_segments = field.split('.')
+      num_segments = len(field_segments)
+      current_map = field_map
+      for i, segment in enumerate(field_segments):
+        if segment:
```

Comment:
Per feedback from the team, I am going to split out `fields[]` into two separate parameters: `include[]` and `exclude[]` such that they can be toggled individually (there seems to be a lot of pushback around exclusions) -- this should not change the underlying map being built though, just the parsing logic